test/library/packages/SharedObject/shared-nil-in-arg.chpl

```diff
@@ -0,0 +1,9 @@
+use SharedObject;
+
+class C { }
+
+proc foo(in sc: Shared(C)) { }
+
+var sc = new Shared(nil:C);
```

Comment:
@mppf, @daviditen - I wonder if we should provide an initializer for Shared that just accepts a type, for convenience. That way the user would write `new Shared(C)` instead of `new Shared(nil:C)`.

Ditto for Unique.

Reply:
@vasslitvinov - you could always write 
``` chapel
var sc:Shared(C);
```

today, couldn't you?

Additionally I'm uncertain what the future will be for Owned and Shared storing `nil` - this is one of the things we might seek to make compile-time checking for. If it's going to be a compile-time error in some situations I'm not sure we're ready to have a shortcut for it.

Reply:
The way I'm using it in the `DateTime` module it needs to be able to store `nil`.  The `TZInfo` class representing the time zone is `nil` for instances that don't want to bother with time zones.  I think this will be the common case, but I want to make it `Shared` for when it is actually used.

Reply:
> you could always write
> 
> `var sc:Shared(C);`
>
> today, couldn't you?

Yes - in fact one of the tests added in this PR did that.